app/Http/Controllers/api/v1/RoomController.php

```diff
@@ -120,65 +120,59 @@ public function start(Room $room, StartJoinMeeting $request)
         $meeting = $room->runningMeeting();
         if (!$meeting) {
             // Create new meeting
-            $meeting              = $room->meetings()->create();
+            $meeting              = new Meeting();
             $meeting->start       = date('Y-m-d H:i:s');
             $meeting->attendeePW  = bin2hex(random_bytes(5));
             $meeting->moderatorPW = bin2hex(random_bytes(5));
+
+            // Basic load balancing, get server with lowest usage
+            $server = Server::lowestUsage();
+
+            // If no server found, throw error
+            if ($server == null) {
+                abort(CustomStatusCodes::NO_SERVER_AVAILABLE, __('app.errors.no_server_available'));
+            }
+
+            $meeting->server()->associate($server);
+            $meeting->room()->associate($room);
             $meeting->save();
 
-            // Try to find server to host meeting
-            while (true) {
-                try {
-                    // TODO implement load balancer here
-
-                    // Find server that is online
-                    $servers = Server::where('status', true)->where('offline', false)->get();
-                    $server  = $servers->random();
-                } catch (InvalidArgumentException $ex) {
-                    // If no server online, room creation failed
-                    $meeting->forceDelete();
-                    abort(CustomStatusCodes::NO_SERVER_AVAILABLE, __('app.errors.no_server_available'));
-                }
+            // Try to start meeting
+            try {
+                $result = $meeting->startMeeting();
+            } // Catch exceptions, e.g. network connection issues
+            catch (\Exception $exception) {
+                // Remove meeting and set server to offline
+                $meeting->forceDelete();
+                $server->offline = true;
+                $server->save();
+                abort(CustomStatusCodes::ROOM_START_FAILED, __('app.errors.room_start'));
+            }
 
-                // Add found server to the meeting
-                $meeting->server()->associate($server);
-                $meeting->save();
+            // Unkown error with the bbb api, that results in meeting not starting
+            if (!isset($result)) {
```

Comment:
Is there an example where the result doesn't gets set but no exception gets thrown?

Reply:
I can't find any cases, so I remove it